### PR TITLE
ip_resolve - Make config variable overridable

### PR DESCRIPTION
**Pull request by brbzull0**
With this change, host resolution order can now be override per transaction if needed. Documentation was also updated to reflect this change.
Test also included.

### CODE REVIEW

File: doc/developer-guide/api/types/TSOverridableConfigKey.en.rst [hunk 1]
```diff
@@ -153,6 +153,7 @@ Enumeration Members
    .. c:macro:: TS_CONFIG_SSL_CLIENT_SNI_POLICY
    .. c:macro:: TS_CONFIG_SSL_CLIENT_PRIVATE_KEY_FILENAME
    .. c:macro:: TS_CONFIG_SSL_CLIENT_CA_CERT_FILENAME
+   .. c:macro:: TS_CONFIG_HTTP_HOST_RESOLUTION_STYLE
```
Comment (SolidWallOfCode) [inline doc/developer-guide/api/types/TSOverridableConfigKey.en.rst:156]: You said you'd changed this.

Comment (brbzull0) [reply doc/developer-guide/api/types/TSOverridableConfigKey.en.rst:156]: my bad! though it's updated in the code, I didn't update the doc properly.

**Comment by brbzull0** [comment]
[approve ci autest]

**Comment by brbzull0** [comment]
[approve ci autest]

**Comment by brbzull0** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**